sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -159,10 +158,7 @@ case class UnresolvedAttribute(nameParts: Seq[String]) extends Attribute with Un
 
   override def toString: String = s"'$name"
 
-  override def sql: String = name match {
-    case ParserUtils.escapedIdentifier(_) | ParserUtils.qualifiedEscapedIdentifier(_, _) => name
```

Comment:
Can we restore this line? It seems to be there for a reason.

One example I found is ```UnresolvedAttribute("`a.b`").sql``` returns ```"`a.b`"``` before, but changes after this change.

Reply:
Ah, O.K. According to the comment of `UnresolvedAttribute.apply`, `UnresolvedAttribute("`a.b`").sql` should return ``` "`a.b`" ```.
```
  /**
   * Creates an [[UnresolvedAttribute]], parsing segments separated by dots ('.').
   */
```
But I think we can't just restore the logic because the following assertion fails.
```
    val attr4 = UnresolvedAttribute("a`b"::"c.d"::Nil)
    assert(attr4.sql === "`a``b`.`c.d`")
```

I also think, the implementation of `UnresolvedAttribute.apply` is wrong.
If, we give ``` "`a.b`" ```, the attribute name should be regarded as ``` `a.b` ``` but it just split by `.` and the name parts are regarded as ``` `a ``` and ``` b` ```.

I have another solution for this issue and open a PR.